src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -251,101 +250,6 @@ doTxHelp(RPC::Context& context, TxArgs const& args)
-    grpc::Status status = grpc::Status::OK;
-    RPC::Status const& error = res.second;
-    TxResult const& result = res.first;
-    // handle errors
-    if (error.toErrorCode() != rpcSUCCESS)
-    {
-        if (error.toErrorCode() == rpcTXN_NOT_FOUND &&
-            result.searchedAll != TxSearched::unknown)
-        {
-            status = {
-                grpc::StatusCode::NOT_FOUND,
-                "txn not found. searched_all = " +
-                    to_string(
-                        (result.searchedAll == TxSearched::all ? "true"
-                                                               : "false"))};
-        }
-        else
-        {
-            if (error.toErrorCode() == rpcTXN_NOT_FOUND)
-                status = {grpc::StatusCode::NOT_FOUND, "txn not found"};
-            else
-                status = {grpc::StatusCode::INTERNAL, error.message()};
-        }
-    }
-    // no errors
-    else if (result.txn)
-    {
-        auto& txn = result.txn;
-
-        std::shared_ptr<STTx const> stTxn = txn->getSTransaction();
-        if (args.binary)
-        {
-            Serializer s = stTxn->getSerializer();
-            response.set_transaction_binary(s.data(), s.size());
-        }
-        else
-        {
-            RPC::convert(*response.mutable_transaction(), stTxn);
-        }
-
-        response.set_hash(context.params.hash());
-
-        auto ledgerIndex = txn->getLedger();
-        response.set_ledger_index(ledgerIndex);
-        if (ledgerIndex)
-        {
-            auto ct =
-                context.app.getLedgerMaster().getCloseTimeBySeq(ledgerIndex);
-            if (ct)
-                response.mutable_date()->set_value(
-                    ct->time_since_epoch().count());
-        }
-
-        RPC::convert(
-            *response.mutable_meta()->mutable_transaction_result(),
-            txn->getResult());
-        response.mutable_meta()->mutable_transaction_result()->set_result(
-            transToken(txn->getResult()));
-
-        // populate binary metadata
-        if (auto blob = std::get_if<Blob>(&result.meta))
-        {
-            assert(args.binary);
-            Slice slice = makeSlice(*blob);
-            response.set_meta_binary(slice.data(), slice.size());
-        }
-        // populate meta data
-        else if (auto m = std::get_if<std::shared_ptr<TxMeta>>(&result.meta))
-        {
-            auto& meta = *m;
-            if (meta)
-            {
-                RPC::convert(*response.mutable_meta(), meta);
-                auto amt =
-                    getDeliveredAmount(context, stTxn, *meta, txn->getLedger());
-                if (amt)
-                {
-                    RPC::convert(
-                        *response.mutable_meta()->mutable_delivered_amount(),
-                        *amt);
-                }
-            }
-        }
-        response.set_validated(result.validated);
-    }
-    return {response, status};
-}
-
 Json::Value
 populateJsonResponse(
```

Comment:
I don't actually know what this looked like before we added these gRPC methods, so take what I say next with a grain of salt.

Should we leave `doTxHelp` and `populateJsonResponse` as their own functions, or move it into `doTx()`. It occurs to me that we may have split this out to enable code reuse between the gRPC and JSON-RPC endpoints. 

Is it worth leaving `doTxHelp` as a helper function?